pkg/service/v0/accounts.go

```diff
@@ -471,3 +485,72 @@ func (s Service) isValidEmail(e string) bool {
 	}
 	return emailRegex.MatchString(e)
 }
+
+const (
+	policyDisableStrongPassword     = "DisableStrongPassword"
+	policyDisablePasswordExpiration = "DisablePasswordExpiration"
+)
+
+func passwordPoliciesValid(policies []string) error {
+	for _, v := range policies {
+		if v != policyDisableStrongPassword && v != policyDisablePasswordExpiration {
+			return fmt.Errorf("invalid password-policy %s", v)
+		}
+	}
+
+	return nil
+}
+
+// validateUpdate takes a update field-mask and validates it against a whitelist of updatable paths.
+// Returns a FieldFilter on success which can be passed to the fieldmask_utils..StructToStruct. An error is returned
+// if the mask tries to update no whitelisted fields.
+//
+// Given an empty or nil mask we assume that the client wants to update all whitelisted fields.
+//
+func validateUpdate(mask *field_mask.FieldMask, updatablePaths map[string]struct{}) (fieldmask_utils.FieldFilterContainer, error) {
+	nop := func(s string) string { return s }
+	// Assume that the client wants to update all updatable path if
+	// no field-mask is given, so we create a mask with all paths
+	if mask == nil || len(mask.Paths) == 0 {
+		paths := make([]string, 0, len(updatablePaths))
+		for fieldName := range updatablePaths {
+			paths = append(paths, fieldName)
+		}
+
+		return fieldmask_utils.MaskFromPaths(paths, nop)
+	}
+
+	// Check that only allowed fields are updated
+	for _, v := range mask.Paths {
+		if _, ok := updatablePaths[v]; !ok {
+			return nil, fmt.Errorf("can not update field %s, either unknown or readonly", v)
+		}
+	}
+
+	return fieldmask_utils.MaskFromPaths(mask.Paths, nop)
+}
+
+// debugLogAccount returns a debug-log event with detailed account-info, and filtered password data
+func (s Service) debugLogAccount(a *proto.Account) *zerolog.Event {
```

Comment:
for debbugging logging the account properties as fields is ok. not recommended for production because properties might collide if they are not scoped by an account object.